### PR TITLE
feat:added login auth and cart management

### PR DESCRIPTION
**Pull request by tushargoyal812**
https://user-images.githubusercontent.com/69681896/160273688-a126ce21-24eb-4caf-a463-e8d8d78c0cba.mp4

https://user-images.githubusercontent.com/69681896/160273696-397b8531-045d-44e3-a104-21cc94133b4e.mp4

### CODE REVIEW

File: src/pages/login/login.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import axios from 'axios'
+import { useEffect, useState } from 'react'
+import { useAuth } from '../../filter-context/auth-context'
+import './login.css'
+import { useNavigate } from 'react-router-dom'
+export const Login=()=>{
+    const navigate=useNavigate()
+    const {auth,setAuth}=useAuth()
+    const [show,setShow]=useState({email: "",
```
Comment (vishalg8454) [inline src/pages/login/login.js:9]: name is not good. Should be something like `[userDetails,setUserDetails]`

File: src/pages/cart/cart.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    <div className="card-main-section flex">
+                        <img src={product.image} className="product-img horizontal-img" alt="product-image"/>
+                        <main className="middle horizontal-middle pl-1 pt-0-5">
+                            <p className="para">{product.title}</p>
+                            <div className="small-1">iPhone 13 (256GB) - Midnight</div>
+                            <div className="small-2">Rs.{product.price}/-
+                                {/* <span className="real-price">MRP: ₹79900(1% OFF)
+                                </span> */}
+                            </div>
+                            <div className="btn-section btn-end flex">
```
Comment (vishalg8454) [inline src/pages/cart/cart.js]: remove comments and logs from the code.

File: src/pages/products/products.js [hunk 3]
```diff
@@ -15,6 +23,41 @@ export const Products=()=>{
     const data6=rating(data5,state)
     const data7=categoryHandler(data6,state)
     const compose=data7
+
+    const [cartName,setCartName]=useState(false)
```
Comment (vishalg8454) [inline src/pages/products/products.js:27]: again the variable naming is not explanatory.

**Review by vishalg8454 — COMMENTED** [review]
A live preview link would have helped verifying the functionality.

File: src/components/nav/nav.js [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
+import { useCart } from '../../filter-context/cart-context'
+import { useAuth } from '../../filter-context/auth-context'
 export const Navbar=()=>{
 
+    const {cartCount}=useCart()
+    const {auth,setAuth}=useAuth()
+
+    const logoutHandler=()=>{
+        localStorage.removeItem("user")
+        // setAuth(false)
```
Comment (deepak29-git) [inline src/components/nav/nav.js:12]: remove unnecessary code

File: src/pages/cart/cart.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        console.log(product._id,product.id,"product");
+        try{
+            const response=await axios.delete(`/api/user/cart/${product._id}`,{
+                headers: {
+                    authorization: token, // passing token as an authorization header
+                  },
+            })
+            setCart(response.data.cart);
+            setCartCount(cartCount=>cartCount-1)
+            console.log(response);
```
Comment (deepak29-git) [inline src/pages/cart/cart.js:23]: remove console

File: src/reducer-func/cart-reducer.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export const cartReducer=(state,action)=>{
+    switch (action.type) {
+        case "add-to-cart":
+            console.log('haa add to cart h');
```
Comment (deepak29-git) [inline src/reducer-func/cart-reducer.js]: delete unused file

**Review by deepak29-git — APPROVED** [review]
your folder structure is good .rest of looks fine

File: src/pages/login/login.js (hunk 1 above)
Comment (tushargoyal812) [reply src/pages/login/login.js:9]: ok will correct it

File: src/pages/cart/cart.js (hunk 2 above)
Comment (tushargoyal812) [reply src/pages/cart/cart.js]: ok

File: src/components/nav/nav.js (hunk 4 above)
Comment (tushargoyal812) [reply src/components/nav/nav.js:12]: ok done

File: src/pages/cart/cart.js (hunk 5 above)
Comment (tushargoyal812) [reply src/pages/cart/cart.js:23]: ok

File: src/reducer-func/cart-reducer.js (hunk 6 above)
Comment (tushargoyal812) [reply src/reducer-func/cart-reducer.js]: ok

**Pull request merged**